### PR TITLE
Improvments

### PR DESCRIPTION
**Pull request by Iliya-usov**

### CODE REVIEW

**Comment by ForNeVeR** [comment]
`System.Threading.Channels` NuGet package depends on `System.Threading.Tasks.Extensions`, which depends on `System.Runtime.CompilerServices.Unsafe`, which I remember as being problematic.
image

File: rd-net/Lifetimes/Lifetimes.csproj [hunk 1]
```diff
@@ -18,6 +18,10 @@
         
     </PropertyGroup>
     
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net35'">
+        <WarningsNotAsErrors>CS8632</WarningsNotAsErrors>
+    </PropertyGroup>
+
```
Comment (ForNeVeR) [inline rd-net/Lifetimes/Lifetimes.csproj:24]: Should this actually go to the `Directory.Build.props`?

Actually wait a minute, our `Directory.Build.props` already has `<NoWarn>nullable</NoWarn>` for this TFM. Doesn't it work?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by ForNeVeR** [comment]
Turns out we have already included all of the `System.Threading.Channels`' dependencies, so I retract my complaint. Using the package is okay.

File: rd-net/Lifetimes/Lifetimes.csproj (hunk 1 above)
Comment (Iliya-usov) [reply rd-net/Lifetimes/Lifetimes.csproj:24]: Probably it doesn't work for Rider.  Rider shows a lot of errors in SWEA.

Comment (ForNeVeR) [reply rd-net/Lifetimes/Lifetimes.csproj:24]: Let's unvestigate; I was unable to reproduce.

**Review by ForNeVeR — COMMENTED** [review]
Sorry, but no. RiderFlow gets broken with this new dependency. We cannot allow us to include `System.Threading.Channels`.

File: rd-net/Lifetimes/Lifetimes.csproj [hunk 2]
```diff
@@ -59,4 +59,8 @@
         <None Include="PackageIcon.png" Pack="true" PackagePath="" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(TargetFramework)' != 'net35'">
+      <PackageReference Include="System.Threading.Channels" Version="6.0.0" />
+    </ItemGroup>
```
Comment (ForNeVeR) [inline rd-net/Lifetimes/Lifetimes.csproj:64]: We don't need this anymore, right?

**Review by ForNeVeR — APPROVED** [review]

File: rd-net/Lifetimes/Lifetimes.csproj (hunk 2 above)
Comment (Iliya-usov) [reply rd-net/Lifetimes/Lifetimes.csproj]: yep

**Pull request merged**